contracts/implementation/PoolCommitter.sol

```diff
@@ -59,301 +48,247 @@ contract PoolCommitter is IPoolCommitter, Ownable {
-                    nextLatestCommitUnexecuted -= 1;
-                    continue;
-                } else {
-                    break;
-                }
-            }
-            if (nextLatestCommitUnexecuted < earliestCommitUnexecuted) {
-                // We have just bumped earliestCommitUnexecuted above latestCommitUnexecuted,
-                // we have therefore run out of commits
-                earliestCommitUnexecuted = NO_COMMITS_REMAINING;
-            } else {
-                latestCommitUnexecuted = nextLatestCommitUnexecuted;
-            }
-        }
+    function executeGivenCommitments(Commitment memory _commits) internal {
+        ILeveragedPool pool = ILeveragedPool(leveragedPool);
+
+        uint256 shortBalance = pool.shortBalance();
+        uint256 longBalance = pool.longBalance();
+        uint256 longTotalSupplyBefore = IERC20(tokens[0]).totalSupply();
+        uint256 shortTotalSupplyBefore = IERC20(tokens[1]).totalSupply();
+        // TODO figure out shadow pool usage
+
+        // Update price before values change
+        priceHistory[updateIntervalId] = Prices({
+            longPrice: PoolSwapLibrary.getPrice(longBalance, longTotalSupplyBefore + _commits.longBurnAmount),
+            shortPrice: PoolSwapLibrary.getPrice(shortBalance, shortTotalSupplyBefore + _commits.shortBurnAmount)
+        });
+
+        // Long Mints
+        uint256 longMintAmount = PoolSwapLibrary.getMintAmount(
+            longTotalSupplyBefore, // long token total supply,
+            _commits.longMintAmount, // amount of quote tokens commited to enter
+            longBalance, // total quote tokens in the long pull
+            _commits.longBurnAmount // total pool tokens commited to be burned
+        );
+        pool.mintTokens(0, longMintAmount, leveragedPool);
+
+        // Long Burns
+        uint256 longBurnAmount = PoolSwapLibrary.getWithdrawAmountOnBurn(
+            longTotalSupplyBefore,
+            _commits.longBurnAmount,
+            longBalance,
+            _commits.longBurnAmount
+        );
+
+        // Short Mints
+        uint256 shortMintAmount = PoolSwapLibrary.getMintAmount(
+            shortTotalSupplyBefore, // short token total supply
+            _commits.shortMintAmount,
+            shortBalance,
+            _commits.shortBurnAmount
+        );
+
+        pool.mintTokens(1, shortMintAmount, leveragedPool);
+
+        // Short Burns
+        uint256 shortBurnAmount = PoolSwapLibrary.getWithdrawAmountOnBurn(
+            shortTotalSupplyBefore,
+            _commits.shortBurnAmount,
+            shortBalance,
+            _commits.shortBurnAmount
+        );
+
+        uint256 newLongBalance = longBalance + _commits.longMintAmount - longBurnAmount;
+        uint256 newShortBalance = shortBalance + _commits.shortMintAmount - shortBurnAmount;
+
+        // TODO update user's aggregate balance
+        // TODO setNewPoolBalances once at end
+        // TODO only mint/burn if amount is > 0
```

Comment:
I understand this is part of a series of changes to be made, but are these intentionally left in or are they stray?